src/schedule/ll_schedule.c

```diff
@@ -200,11 +201,14 @@ static int schedule_ll_domain_set(struct ll_schedule_data *sch,
 
 	spin_lock(&sch->domain->lock);
 
+	registered = sch->domain->registered[core];
 	if (atomic_add(&sch->num_tasks, 1) == 1)
 		sch->domain->registered[core] = true;
 
-	if (atomic_add(&sch->domain->total_num_tasks, 1) == 1) {
+	if (atomic_add(&sch->domain->total_num_tasks, 1) == 1)
 		domain_set(sch->domain, platform_timer_get(timer_get()));
+
+	if (!registered) {
```

Comment:
Looks fine to me, I would just comment the logic here so that it's obvious why we are doing this